### PR TITLE
Fix pylint-odoo and click-odoo installation

### PR DESCRIPTION
**Pull request by joao-p-marques**
~~It is required for the package setup https://github.com/OCA/pylint-odoo/blob/master/setup.py#L5
But not specified in the requirements, and pip has no control on its insallation. It is done with easy_install, which may cause problems.
See https://pip.pypa.io/en/latest/cli/pip_install/#controlling-setup-requires~~

Install version from OCA/pylint-odoo#329 with a more modern packaging method, avoiding errors with setuptools

### CODE REVIEW

**Comment by ap-wtioit** [comment]
I also hit the pbr installation problem on my local system today:
```
build 1/3 Step 19/69 : RUN python -m venv --system-site-packages /qa/venv     && . /qa/venv/bin/activate     && pip install --no-cache-dir         click         coverage         flake8         pylint-odoo         six     && npm install --loglevel error --prefix /qa 'eslint@<7'     && deactivate     && mkdir -p /qa/artifacts     && git clone --depth 1 $MQT /qa/mqt
build 1/3  ---> Running in 1511a0461902
build 1/3 Requirement already satisfied: click in /usr/local/lib/python3.5/site-packages
build 1/3 Collecting coverage
build 1/3   Downloading https://files.pythonhosted.org/packages/18/97/fda94f1e78b508dd1a493106c236076c4c82a9925c9347211afc4d6bd4e0/coverage-5.5-cp35-cp35m-manylinux1_x86_64.whl (240kB)
build 1/3 Collecting flake8
build 1/3   Downloading https://files.pythonhosted.org/packages/a0/b0/3b5820728d687f2c000476216a3fccc7a03baac1034afc0284ccde25e26d/flake8-3.9.1-py2.py3-none-any.whl (73kB)
build 1/3 Collecting pylint-odoo
build 1/3   Downloading https://files.pythonhosted.org/packages/74/ae/9cc3d387291e1e2866e0af906e1b76e766885867fe765a3f296f5879fd42/pylint-odoo-3.7.1.tar.gz (69kB)
build 1/3     Complete output from command python setup.py egg_info:
build 1/3     Couldn't find index page for 'pbr' (maybe misspelled?)
build 1/3     No local packages or working download links found for pbr
build 1/3     Traceback (most recent call last):
build 1/3       File "<string>", line 1, in <module>
build 1/3       File "/tmp/pip-build-ac0ljojy/pylint-odoo/setup.py", line 8, in <module>
build 1/3         package_data={'': ['*.yaml']})
build 1/3       File "/usr/local/lib/python3.5/distutils/core.py", line 108, in setup
build 1/3         _setup_distribution = dist = klass(attrs)
build 1/3       File "/qa/venv/lib/python3.5/site-packages/setuptools/dist.py", line 315, in __init__
build 1/3         self.fetch_build_eggs(attrs['setup_requires'])
build 1/3       File "/qa/venv/lib/python3.5/site-packages/setuptools/dist.py", line 361, in fetch_build_eggs
build 1/3         replace_conflicting=True,
build 1/3       File "/qa/venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 850, in resolve
build 1/3         dist = best[req.key] = env.best_match(req, ws, installer)
build 1/3       File "/qa/venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 1122, in best_match
build 1/3         return self.obtain(req, installer)
build 1/3       File "/qa/venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 1134, in obtain
build 1/3         return installer(requirement)
build 1/3       File "/qa/venv/lib/python3.5/site-packages/setuptools/dist.py", line 429, in fetch_build_egg
build 1/3         return cmd.easy_install(req)
build 1/3       File "/qa/venv/lib/python3.5/site-packages/setuptools/command/easy_install.py", line 659, in easy_install
build 1/3         raise DistutilsError(msg)
build 1/3     distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pbr')
```

**Comment by joao-p-marques** [comment]
Yes, maybe we should merge this until this is fixed upstream. WDYT @Yajo?

File: 11.0.Dockerfile [hunk 1]
```diff
@@ -114,11 +114,13 @@ RUN mkdir -p auto/addons auto/geoip custom/src/private \
...
 RUN python -m venv --system-site-packages /qa/venv \
     && . /qa/venv/bin/activate \
+    # HACK: Upgrade pip: higher version needed to install pyproject.toml based packages
+    && pip install -U pip \
     && pip install --no-cache-dir \
         click \
         coverage \
         flake8 \
-        pylint-odoo \
+        git+https://github.com/Tecnativa/pylint-odoo.git@migrate-poetry \
```
Comment (Yajo) [inline 11.0.Dockerfile:123]: Please change this, so it is a better reminder that tells us this should be undone when that PR is merged:

```suggestion
        git+https://github.com/OCA/pylint-odoo.git@refs/pull/329/head \
```

File: 12.0.Dockerfile [hunk 2]
```diff
@@ -107,11 +107,13 @@ RUN mkdir -p auto/addons auto/geoip custom/src/private \
...
 RUN python -m venv --system-site-packages /qa/venv \
     && . /qa/venv/bin/activate \
+    # HACK: Upgrade pip: higher version needed to install pyproject.toml based packages
+    && pip install -U pip \
     && pip install --no-cache-dir \
         click \
         coverage \
         flake8 \
-        pylint-odoo \
+        git+https://github.com/Tecnativa/pylint-odoo.git@migrate-poetry \
```
Comment (Yajo) [inline 12.0.Dockerfile:116]:
```suggestion
        git+https://github.com/OCA/pylint-odoo.git@refs/pull/329/head \
```

**Review by Yajo — CHANGES_REQUESTED** [review]
Yes, this shall be merged ASAP. Upstream isn't being responsive.

Just a tiny fix please, that helps our future us to remember why this is here.

**Comment by joao-p-marques** [comment]
Ok @Yajo I also added here the fix from https://github.com/acsone/click-odoo-contrib/pull/93 Or do you prefer that in a separate PR?

**Review by Yajo — APPROVED** [review]
It's perfect, thanks.

**Comment by joao-p-marques** [comment]
@Yajo had to install `click-odoo-contrib` from our fork instead, because of some errors with the version name being picked up by pip:
https://github.com/Tecnativa/doodba/pull/414/checks?check_run_id=2388250709#step:7:3296
https://github.com/acsone/click-odoo-contrib/releases/tag/13.0-slg-dca55aae4-click-odoo-contrib

I left a comment referencing the PR instead.

**Pull request merged**